src/entities/activity/ui/items/RequestHealthRecordDataItem/OneUpHealthStep.tsx

```diff
@@ -29,14 +52,177 @@ export const OneUpHealthStep: FC = () => {
   const submitId = groupProgress?.submitId;
 
   // Fetch the token using our custom hook
-  const { data, isLoading } = useOneUpHealthTokenQuery({ appletId, submitId });
-  const accessToken = data?.data?.result?.accessToken;
+  const { data, isLoading, error } = useOneUpHealthTokenQuery(
+    { appletId, submitId, activityId },
+    { refetchOnWindowFocus: false },
```

Comment:
We should set a stale time here, otherwise we throw away the cached value when the query refetches (and fails) on component re-mount. You probably know a more appropriate value, I just put `60000` as a placeholder
```suggestion
    { refetchOnWindowFocus: false, staleTime: 60000 },
```

https://github.com/user-attachments/assets/cd117146-e150-4786-9e12-424af4e8c84a

---

The query refetch fails because the BE tries to recreate the 1Up user (applet + activity + submit ID), but they already exist so it fails. Not sure why it does this

Reply:
Indeed. I can go ahead and a couple changes on this to improve this functionality. What do you think @sultanofcardio @farmerpaul?

Reply:
@andrevitalb Yes, I think that would be good